### PR TITLE
ep1c12.cpp,cv1k.cpp: More accurate CV1000 Blitter timings. Remove unsafe mode.

### PR DESCRIPTION
**Pull request by buffis**
- Implement cv1000 blitter delay timing for Upload and Draw operations as described in my research here:  https://buffis.com/docs/CV1000_Blitter_Research_by_buffi.pdf . All timing data is analyzed by me.
- Remove epic12 "unsafe" Blitter mode. The delay introduced here doesn't really have much to do with how things behave on proper hardware, and trying to tune it doesn't make a lot of sense.
- Remove some old commented out lines in epic12.cpp which seem to not add much value.
- If EP1C_PRINT_BLITTER_DELAYS is set, print out when a Blitter delay triggers.
- Add documentation for 6 unhandled Blitter command offsets (see blitter_w). Thanks rtw for looking at these.
- Update some docs in cv1k.cpp

This is still not perfect, and further work is needed on CV1000 slowdowns, especially on SH-3 CPU waitstates. I think this is fairly accurate in terms of Blitter timings though.

### CODE REVIEW

**Comment by DavidHaywood** [comment]
FWIW I'm alright with this.

Solving the SH3 timing issue properly will be more difficult, because proper cache emulation / waitstates etc. will require turning off the recomplier, and make the overall emulation much slower.

Having the blitter timings as an error factor rules out is helpful though, and that 'unsafe' code was super ugly.

**Comment by buffis** [comment]
Yes, the SH3 timing is more complicated.
The caching behavior is actually fairly well described in the datasheet, but having it hooked up properly seems complicated, even without the dynamic recompilation.

Unfortunately most of the slowdown in the games I've looked at seems more driven by CPU than Blitter, but Blitter is definitely also a factor.

Using CPU underclock is not an _awful_ replacement for proper wait state insertion, but it will definitely not be accurate.

**Comment by buffis** [comment]
FYI, I'd hold off a bit about merging this.
After more testing, a person notified me that there seems to be unexpected amounts of slowdown in some areas in Pink Sweets, and I don't have a good explanation so far.

I need to investigate that a bit more and figure out if I missed some edge case. Hoping to have time to look at that soon. Overall this code should mirror the Blitter behavior quite well, but feels like some edge case handling is missing.

**Comment by buffis** [comment]
Ok, so did quite a few changes to the branch (see commit above).
For clipping, there's some info here:
https://buffis.com/research/more-cv1000-research-now-featuring-clipping/

I'll try to get some more people testing the branch to see if I can find any other issues with this, but overall this should be quite close.

**Comment by buffis** [comment]
So uh, how do you want to proceed with this? Would be neat to merge it in

**Comment by buffis** [comment]
Ehrm, friendly ping? Anyone want to have a look at this?

**Comment by cuavas** [comment]
I’ll get to it.

File: src/devices/video/epic12.cpp
Comment (cuavas) [inline src/devices/video/epic12.cpp:801]: Please avoid using the preprocessor like this – the logging code will rot.  You can use the configurable logging from logmacro.h (the compiler is smart enough to optimise it away if logging is disabled).

**Review by cuavas — CHANGES_REQUESTED** [review]
Most of the stuff is fairly minor.  My biggest concern is the static data member masquerading as an instance member – that’s usually a bad sign.

File: src/devices/video/epic12.h
Comment (buffis) [reply src/devices/video/epic12.h]: Made static inline constexpr protected member

File: src/devices/video/epic12.h
Comment (buffis) [reply src/devices/video/epic12.h]: I think this is from some earlier version of my code, thanks for catching

File: src/devices/video/epic12.h
Comment (buffis) [reply src/devices/video/epic12.h]: Made instance member, dont remember why I had this static. Member is fine

File: src/devices/video/epic12.cpp
Comment (buffis) [reply src/devices/video/epic12.cpp]: I think the way I changed it to is correct?
Please verify. I'm not used to Mame libraries.

**Comment by buffis** [comment]
Comments from review is are addressed. There's also a merge confict in epic12.cpp that someone with write access will need to resolve.

The code that's changed is removed in my code, since that profiling no longer makes sense, so just make sure that the profiler lines are stripped out.

Thanks!

**Comment by cuavas** [comment]
OK, I can take it from here.

**Comment by cuavas** [comment]
I’d advise you to reset your `master` branch onto the upstream `master` branch, as I’ve left a bit of a mess on it.

**Pull request merged**
